src/contrast.rs

```diff
@@ -7,9 +8,86 @@ use image::{
     Luma
 };
 
-/// Returns the cumulative histogram of grayscale values in an 8bpp
+/// Returns the Otsu threshold level of an 8bpp image.
+/// This threshold will optimally binarize an image that
+/// meets certain criteria.
+pub fn otsu_level<I>(image: &I) -> u8
+    where I: GenericImage<Pixel=Luma<u8>> {
+    let hist = histogram(image);
+    let levels = hist.len();
+    let mut histc = [0i32; 256];
+    let mut meanc = [0.0f64; 256];
+    let mut pixel_count = hist[0] as f64;
+
+
+    histc[0] = hist[0];
+
+    for i in 1..levels {
+        pixel_count += hist[i] as f64;
+        histc[i] = histc[i-1] + hist[i];
+        meanc[i] = meanc[i-1] + (hist[i] as f64)*(i as f64);
+    }
+
+    let mut sigma_max:f64 = -1.0;
+    let mut otsu:f64 = 0.0;
+    let mut otsu2:f64 = 0.0;
+
```

Comment:
Minor nitpick: most rust uses a space after the colons here. Also, you could just write let mut x = 0f64.